controller/main.go

```diff
@@ -102,14 +102,17 @@ func (c *controller) SetPools(l log.Logger, pools map[string]*config.Pool) contr
 	level.Debug(l).Log("event", "startUpdate", "msg", "start of config update")
 	defer level.Debug(l).Log("event", "endUpdate", "msg", "end of config update")
 
-	if len(pools) == 0 {
+	if pools == nil {
 		level.Error(l).Log("op", "setConfig", "error", "no MetalLB configuration in cluster", "msg", "configuration is missing, MetalLB will not function")
 		return controllers.SyncStateErrorNoRetry
 	}
 
-	if err := c.ips.SetPools(pools); err != nil {
-		level.Error(l).Log("op", "setConfig", "error", err, "msg", "applying new configuration failed")
-		return controllers.SyncStateError
+	if len(pools) != 0 {
```

Comment:
Why do you need this check? This is changing the previous behaviour I guess

Reply:
yes. that's why I looked at v0.12 to see if it's happened there too.

Reply:
and without this check, the case of calling SetPools with empty pools would
return SyncStateError

Reply:
Not sure about that, I think it would have returned SyncStateReprocessAll
See https://github.com/metallb/metallb/blob/f691c94622a5ab0aab29511df9911cf7bf1f94ac/internal/allocator/allocator.go#L66 , that will return nil

Reply:
I saw it and poolFor is returning "" for empty pools resulting of
returning an error

Reply:
Right, but that makes sense. If the pool is allocated to some service we want to fail

Reply:
right. but it will fail anyway even without the service being allocated.

Reply:
if a service is not allocated, a.allocated will be empty and poolFor won't be called? What am I missing?

Reply:
to resolve this conversation, you were right, shouldn't add this check.